### ISSUE TITLE
not webserver agnostic

### ISSUE THREAD

**Issue by mckaygerhard** [kishor10d/Admin-Panel-User-Management-using-CodeIgniter#105]
the app only works with rewrite rule activated that only documented for apache2 in htaccess

**Comment by kishor10d** [comment]
@mckaygerhard I will add rewrite rules for others as well (e.g. lighthttpd)

**Comment by mckaygerhard** [comment]
@kishor10d  there is the lighttpd rules ? why closed if there is no commit related?

**Comment by mckaygerhard** [comment]
to avoid the rewrite nasty dependent you must change some ocurrences of codeigniter of `site_url()` to `base_url()` 

by example to call public files and assets you must use `base_url()` that is the site url without the index defined (so will not be afected if you use rewrite rule or not)

for the hidden urls you must setup a  `site_url()` with detection of the rewrite, by a flag .. this is the right way how good project like **dokuwiki** does!


another more elegant way could be a class/helper that detects if the site have a index,php defined or not, and return proper result from `site_url()` or `base_url()`  depending of the detection..  for CEO urls 

**i can just started to find ways to made such change but of course only if you will also give enought support...** 

cos i was xpecting to use this repo as template to **a future auth api** .. a **missing kind of artifact in codeigniter**.. , currently  only you can find libs like Ion auth, myth-auth or the new hyper complicated shield.. all of those are just libraires that you must code.. my project will be an API that just will be consumed and coding will be only to provide more funtionallity @kishor10d

**Comment by gstrauss** [comment]
lighttpd config suggestion
```
index-file.names = ("index.html", "index.php")

server.modules += ("mod_rewrite")
$HTTP["url"] !~ "^/(?:forums|images|css|downloads|jquery|js|robots\.txt|favicon\.ico|.*\.swf)|/$" {
    url.rewrite-if-not-file = ("" => "/index.php?${url.path}${qsa}")
}
```

**Comment by mckaygerhard** [comment]
Why would @gstrauss will be interested in what "_**old applications**_" run in his project? that ultimately leaves a lot behind and has little welcome in my latest suggestions? lighttps is not used so much? hug?

```
$HTTP["url"] !~ "^/(?:forums|images|css|downloads|jquery|js|robots\.txt|favicon\.ico|.*\.swf)|/$" {
    url.rewrite-if-not-file = ("" => "/index.php?${url.path}${qsa}")
}
```

i will check this @kishor10d  but just advertise that index.php depends of the defined in config.php file, must be pointed in the configuration file, so if changed still must be recodified the usage of `site_url()` to `base_url()` but i am still busy ..

**Comment by muhammadasad251** [comment]
I want to use this application on my server. I put it in my subdirectory and Logme controller is not showing anything. When I press login button

**Comment by kishor10d** [comment]
@muhammadasad251 : As per my understanding, you need to change the base url in config.php file.

**Comment by mckaygerhard** [comment]
that's not true.. this web application forced to use rewrite.. making server focused and not agnostic, i already put the rule rewrite you  provide it and dont workt.. 

advanced sistems like dockuwiki already does this whitout forced admins to use apache2 @muhammadasad251
